crates/polars-row/src/row.rs

```diff
@@ -87,8 +93,13 @@ impl RowsEncoded {
 
         unsafe {
             let (_, values, _) = mmap::slice(&self.values).into_inner();
-            let offsets = bytemuck::cast_slice::<usize, i64>(self.offsets.as_slice());
-            let (_, offsets, _) = mmap::slice(offsets).into_inner();
+            let offsets = match bytemuck::try_cast_slice::<usize, i64>(self.offsets.as_slice()) {
```

Comment:
I think we should then just keep a `Vec<u64>` instead of a `Vec<usize>`.

Reply:
Hi @ritchie46 , I updated the code, I am not all too happy about it because I introduced lots of casting from u64 to usize and vice versa. Imo there is more significant refactoring needed to reduce the casting - by abstracting the `RowsEncoded` such that the other structs are no longer able to directly write on the offset vector. If you agree with that, I can do it, let me know.

Also, on more point, the WASM build also fails on `utf8_to` module. I misled the compiler, but it isn't a real fix. Please take a look at that part as well.